src/store/index.ts

```diff
@@ -54,41 +64,73 @@ export const resultsFilter = ref<ResultsFilter>({
   },
 });
 
-export const addFootprint = (spatialId: number) => {
-  if (footprintIds.value === undefined) {
-    footprintIds.value = [];
+export const addFootprint = async (spatialId: number, region?: boolean) => {
+  let key;
+  let footprint;
+  if (!region) {
+    footprint = (await rgdFootprint(spatialId)).footprint;
+    key = `result_${spatialId}`;
+  } else {
+    footprint = regionsList.value?.find((reg) => reg.id === spatialId)?.footprint;
+    key = `region_${spatialId}`;
+  }
+  if (key && footprint) {
+    visibleFootprints.value = { [key]: footprint, ...visibleFootprints.value };
   }
-  footprintIds.value.push(spatialId);
 };
 
-export const removeFootprint = (spatialId: number) => {
-  footprintIds.value = footprintIds.value.filter((obj: number) => obj !== spatialId);
+export const removeFootprint = (spatialId: number, region?: boolean) => {
+  let key: string;
+  if (!region) {
+    key = `result_${spatialId}`;
+  } else {
+    key = `region_${spatialId}`;
+  }
+  if (visibleFootprints.value[key]) {
+    visibleFootprints.value = Object.fromEntries(
+      Object.entries(visibleFootprints.value).filter(([k]) => k !== key),
+    );
+  }
 };
 
-export const addVisibleOverlay = (spatialId: number) => {
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const addVisibleOverlay = (spatialId: number, region?: boolean) => {
   if (visibleOverlayIds.value === undefined) {
     visibleOverlayIds.value = [];
   }
   visibleOverlayIds.value.push(spatialId);
 };
 
-export const removeVisibleOverlay = (spatialId: number) => {
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const removeVisibleOverlay = (spatialId: number, region?: boolean) => {
   visibleOverlayIds.value = visibleOverlayIds.value.filter((obj: number) => obj !== spatialId);
 };
 
-export const selectResultForMetadataDrawer = async (spatialId: number) => {
+export const selectResultForMetadataDrawer = async (spatialId: number, region?: boolean) => {
   if (searchResults.value) {
     searchResults.value = searchResults.value.map(
       // eslint-disable-next-line @typescript-eslint/camelcase
       (entry) => Object.assign(entry, { show_metadata: false }),
     );
+    if (!region) {
+      const res = await rgdImagery(spatialId);
+      drawerContents.value = res;
+    }
+  } if (regionsList.value) {
+    regionsList.value = regionsList.value.map(
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      (entry) => Object.assign(entry, { show_metadata: false }),
+    );
+    if (region) {
+      drawerContents.value = regionsList.value.filter(
+        (entry) => entry.id === spatialId,
+      )[0].properties;
```

Comment:
I think you want [Array.prototype.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) instead.